qutebrowser/mainwindow/tabwidget.py

```diff
@@ -439,11 +445,24 @@ def tabSizeHint(self, index):
             # get scroll buttons as soon as needed.
             size = minimum_size
         else:
+            #TODO: relative size and/or configured one
+            try:
+                tab = objreg.get('tab', scope='tab', window=self._win_id, tab=index)
```

Comment:
You'd probably need to handle this via `tab_data` and `set_tab_data` (which is unrelated to the `tab.data` I mentioned above) - see how it's handled for `indicator-color` for example.